ERCS/eip-7281.md

```diff
@@ -0,0 +1,362 @@
+4. Allowlist all desired bridges *including* the canonical bridge that owns the legacy implementation, mapping the xERC-20 representation on the home domain to the xERC-20 representation on the remote.
+5. At this point, it is fully possible for users to begin transferring the token across all bridges as would be expected from the xERC-20 standard. However, note that the canonical bridge connecting home to remote will now have two bridge paths: (a) ERC-20→ERC-20 (locking the ERC-20 at home), and (b) xERC-20→xERC-20. Now, the issuer can begin the process of sunsetting the legacy canonical ERC-20 on the remote domain.
+6. The issuer should at this point disallow minting *new* legacy canonical ERC-20s on the remote domain. This would organically & gradually lead to the legacy ERC-20s on the remote chain to become locked into the lockbox.
+7. At any point in the future, the token issuer can then use their own treasury to unwrap xERC-20s on the remote domain into legacy ERC-20s and send them back to the home chain, incurring the latency of this on behalf of users.
+
+### Compatibility with Bridges
+
+One key benefit to the xERC-20 model is that the simple requirement of a burn & mint interface makes this proposal compatible with most bridges either out of the box or through a predefined custom token mapping process. The following bridges & domains were considered in the creation of this proposal:
+
+- **3rd party bridges:** every 3rd party bridge that relies on messaging infrastructure already supports a mint and burn interface and has a pathway to map custom minted tokens.
+- **Arbitrum rollup bridge:**  Arbitrum allows token issuers to [write a custom gateway](https://developer.arbitrum.io/asset-bridging#other-flavors-of-gateways).
+- **OP Stack bridge:** Optimism supports [adding a custom bridge](https://community.optimism.io/docs/guides/bridge-dev/#building-a-custom-bridge).
+- **Polygon PoS bridge:** Polygon supports custom tokens [through their fx portal](https://wiki.polygon.technology/docs/develop/l1-l2-communication/fx-portal).
+- **ZkSync bridge**: Zksync supports [custom bridges of any kind](https://era.zksync.io/docs/reference/concepts/bridging/bridging-asset.html#custom-bridges-on-l1-and-l2).
+- **GnosisChain Omnibridge:** GnosisChain does not currently support custom tokens, but is in the process of redesigning their omnibridge and plans to allow this functionality.
+
+Note: In most of the above cases, a custom bridge integration also means integration into canonical bridge UIs, ensuring that users have a consistent experience throughout the process. Additionally, a single xERC-20 custom bridge implementation could be built for each ecosystem and serve any number of token issuers.
+
+## Reference Implementation
+
+You can find a reference implementation and associated test cases [here](https://github.com/defi-wonderland/xTokens/blob/dev/solidity/contracts/XERC-20.sol).
```

Comment:
You'll need to either remove this section, or include a compatibly licensed (preferably CC0-1.0) reference implementation in your assets directory.